src/framework/linalg/almost_equal.hpp

```diff
@@ -29,7 +29,7 @@ namespace Linalg {
 // If we have numbers closer to 0, then max_diff can be set to a value
 // way smaller than epsilon. For numbers larger than 1.0, epsilon will
 // scale (the bigger the number, the bigger the epsilon).
-template <typename T, typename = enable_if_numeric_t<T>>
```

Comment:
This function was specifically implemented to add complex to integral constant check (you don't really need to define an enable_if_numeric because it's basically a built in).  There are a lot of other areas of code you will need to change if you have modified this in all the linalg functions.

Reply:
Ok, let's keep it untouched because as you said is being used through all the linalg library, but I'm not sure that this will have consequences for functions that need specialization for complex numbers... maybe it's just semantics, or maybe these functions will just fail compiling at some point (which is acceptable).